packages/edit-post/src/components/visual-editor/index.js

```diff
@@ -183,25 +205,50 @@ export default function VisualEditor( { styles } ) {
 
 	const blockSelectionClearerRef = useBlockSelectionClearer();
 
-	const layout = useMemo( () => {
+	// fallbackLayout is used if there is no Post Content,
+	// and for Post Title.
+	const fallbackLayout = useMemo( () => {
 		if ( isTemplateMode ) {
 			return { type: 'default' };
 		}
 
 		if ( themeSupportsLayout ) {
 			// We need to ensure support for wide and full alignments,
 			// so we add the constrained type.
-			return { ...defaultLayout, type: 'constrained' };
+			return { ...globalLayoutSettings, type: 'constrained' };
 		}
 		// Set default layout for classic themes so all alignments are supported.
 		return { type: 'default' };
-	}, [ isTemplateMode, themeSupportsLayout, defaultLayout ] );
+	}, [ isTemplateMode, themeSupportsLayout, globalLayoutSettings ] );
+
+	const postContentBlock = useMemo( () => {
+		return findPostContent( parse( templateContent ) );
+	}, [ templateContent ] );
 
-	const blockListLayoutClass = classnames( {
-		'is-layout-constrained': themeSupportsLayout,
-		'is-layout-flow': ! themeSupportsLayout,
-		'has-global-padding': useRootPaddingAwareAlignments,
-	} );
+	const postContentLayoutClasses = useLayoutClasses(
+		postContentBlock?.attributes?.layout,
+		globalLayoutSettings?.definitions
+	);
+
+	const blockListLayoutClass = classnames(
+		{
+			'is-layout-flow': ! themeSupportsLayout,
+			'wp-container-visual-editor': themeSupportsLayout,
+		},
+		postContentLayoutClasses
+	);
+
+	const postContentLayoutStyles = useLayoutStyles(
+		postContentBlock,
+		'.wp-container-visual-editor'
+	);
+
+	// If there is a Post Content block we use its layout, or 'default' layout
+	// if it doesn't have one. If no Post Content this must be a classic theme,
+	// in which case we use the fallback layout.
+	const postContentLayout = postContentBlock
+		? postContentBlock?.attributes?.layout || { type: 'default' }
```

Comment:
For this one, do we need the fallback logic for legacy markup (e.g. `inherit`) here, too? I noticed in TwentyTwentyTwo, the `single` template appears to still be using `inherit` so the view in the post editor for me was full width because it was assuming `default` (flow) layout instead of `constrained`:

![image](https://user-images.githubusercontent.com/14988353/191408980-d841e9c4-d66d-4052-903e-d1ada78000f5.png)

Reply:
Ooooh good point! On it.